packages/vite/src/node/plugins/worker.ts

```diff
@@ -341,7 +341,7 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
               objURL = blob && (self.URL || self.webkitURL).createObjectURL(blob);
               if (!objURL) throw ''
               const worker = new ${workerConstructor}(objURL, ${workerTypeOption});
-              worker.addEventListener("error", () => {
+              worker.addEventListener("message", () => {
```

Comment:
We should call `revokeObjectURL` if an error happened regardless of whether a message has been received because an error may happen before a message is received (or the user may not send any messages).
We need to append `revokeObjectURL` call to the script so it is called even if the user didn't send any messages (for ESM it's done above in the declaration of `blob` variable).